python/cudf/cudf/core/column/timedelta.py

```diff
@@ -0,0 +1,402 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+import datetime as dt
+from numbers import Number
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+
+import cudf
+from cudf import _lib as libcudf
+from cudf._lib.nvtx import annotate
+from cudf._lib.scalar import Scalar, as_scalar
+from cudf.core.buffer import Buffer
+from cudf.core.column import column
+from cudf.core.column.datetime import _numpy_to_pandas_conversion
+from cudf.utils.dtypes import is_scalar, np_to_pa_dtype
+
+
+class TimeDeltaColumn(column.ColumnBase):
+    def __init__(
+        self, data, dtype, mask=None, size=None, offset=0, null_count=None
+    ):
+        """
+        Parameters
+        ----------
+        data : Buffer
+            The Timedelta values
+        dtype : np.dtype
+            The data type
+        mask : Buffer; optional
+            The validity mask
+        size : int, optional
+            Size of memory allocation.
+        offset : int
+            Data offset
+        null_count : int, optional
+            The number of null values.
+            If None, it is calculated automatically.
+        """
+        dtype = np.dtype(dtype)
+        if data.size % dtype.itemsize:
+            raise ValueError("Buffer size must be divisible by element size")
+        if size is None:
```

Comment:
In general, we are moving to drop the assumption that the `size` can be computed from the underlying allocations (for any Column type). Let's make `size` a required parameter instead.

Reply:
Sure, should we also mandate `size` parameter for the rest of the columns too?